rfd/0073-kube-secret-storage.md

```diff
@@ -0,0 +1,501 @@
+                                                                   │                                                 │                    │                          
+                                                                   │                                                 │                    │                          
+                                                    ╔═══════╤══════╪═════════════════════════════════════════════════╪════════════════════╪═══════════════╗          
+                                                    ║ LOOP  │  key rotation                                          │                    │               ║          
+                                                    ╟───────┘      │                                                 │                    │               ║          
+                                                    ║              │              Rotate certificates                │                    │               ║          
+                                                    ║              │────────────────────────────────────────────────>│                    │               ║          
+                                                    ║              │                                                 │                    │               ║          
+                                                    ║              │                New certificates                 │                    │               ║          
+                                                    ║              │<─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ │                    │               ║          
+                                                    ║              │                                                 │                    │               ║          
+                                                    ║              │                     Update secret content       │                    │               ║          
+                                                    ║              │─────────────────────────────────────────────────────────────────────>│               ║          
+                                                    ╚══════════════╪═════════════════════════════════════════════════╪════════════════════╪═══════════════╝          
+                                                              ┌────┴────┐                                        ┌───┴────┐          ┌────┴─────┐                    
+                                                              │KubeAgent│                                        │Teleport│          │Kubernetes│                    
+                                                              └─────────┘                                        └────────┘          └──────────┘                    
+```
+
+### Backend storage
+
+Kubernetes Secret backend storage must satisfy the `Backend` interface.
```

Comment:
I'm not sure why we need full backend interface tbh. We're not going to use Kubernetes secrets as actual backend for Teleport data, right?

Reply:
The function responsible for writing the agent's identity, `WriteIdentity`, is defined in the following structure:
`go
// ProcessStorage is a backend for local process state,
// it helps to manage rotation for certificate authorities
// and keeps local process credentials - x509 and SSH certs and keys.
type ProcessStorage struct {
	backend.Backend
}
func (p *ProcessStorage) WriteIdentity(name string, id Identity) error {
...
}
` 

`backend.Backend` is the interface mentioned in the doc. 

Probably for this case, we can use a smaller interface in `ProcessStorage`  since it only references `Create, Get, Put, Close`.

```go
type ProcessStorageInterface  interface {
	// Create creates item if it does not exist
	Create(ctx context.Context, i backend.Item) (*backend.Lease, error)
	// Put puts value into backend (creates if it does not
	// exists, updates it otherwise)
	Put(ctx context.Context, i backend.Item) (*backend.Lease, error)
	// Get returns a single item or not found error
	Get(ctx context.Context, key []byte) (*backend.Item, error)
	// Close closes backend and all associated resources
	Close() error
}
```